src/models/datastore.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright 2015 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+let ModelFactory = require('../helpers/model-factory');
+let Connection = require('./connection');
+
+class DataStore {
+
+  constructor () {
+    this.storeName_ = null;
+  }
+
+  configure (opts) {
+    if (typeof opts === 'undefined') {
+      throw new Error('DataStore configuration not provided.');
+    }
+
+    this.storeName = opts.store || null;
+    ModelFactory.modelData = opts.models || null;
+  }
+
+  get storeName () {
+    return this.storeName_;
+  }
+
+  set storeName (storeName) {
+    this.storeName_ = storeName;
+  }
+
+  get models () {
+    return ModelFactory.modelData;
+  }
+
+  set models (modelData) {
+    ModelFactory.modelData = modelData;
+  }
+
+  _ensureValidPopulations (populations) {
+    // Assume missing.
+    if (typeof populations === 'undefined' || populations === null) {
+      return [];
+    }
+
+    // Assume string.
+    if (typeof populations === 'string') {
+      return [populations];
+    }
+
+    // Assume object.
+    if (typeof populations.length === 'undefined') {
+      return [populations];
+    }
+
+    // Assume array.
+    return populations;
+  }
+
+  action_ (type, cb) {
```

Comment:
Maybe a comment to explain that this is a helper method to create connections, perform an action on a model instance and then close the connection.

Throughout reading this code I didn't get what action did each time until about fourth or fifth time.